osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -284,10 +331,32 @@ private void load(OsuColour colours, IBindable<RulesetInfo> ruleset, AudioManage
             Ruleset.BindTo(ruleset);
             if (mods != null) SelectedMods.BindTo(mods);
 
+            SelectedMods.ValueChanged += updateModSettings;
```

Comment:
This should probably be a `BindValueChanged(, true)`.

Also this shouldn't be inside `load()`.

Also this should probably not even exist, and `updateModSettings()` should be invoked by `selectedModsChanged` (see the existing `LoadComplete()`).